File: Source/Mobile/PVScan.Mobile/Models/Sorting.cs
using System;
namespace PVScan.Mobile.Models
{
    public enum SortingField
    {
        None,
        Date,
        Text,
        Format
    };

    public class Sorting
    {
        public SortingField Field { get; set; }
        public bool Descending { get; set; }

        public static Sorting Default() => new Sorting()
        {
            Field = SortingField.Date,
            Descending = true,
        };
    }
}

File: Source/Web/Frontend/app/src/models/Sorting.ts
export enum SortingField {
    None,
    Date,
    Text,
    Format
}

export interface Sorting {
    Field: SortingField;
    Descending: boolean;
}

export function IsDefaultSorting(sorting: Sorting): boolean {
    return (sorting.Descending &&
            sorting.Field == SortingField.Date);
}

export function DefaultSorting(): Sorting {
    return {
        Descending: true,
        Field: SortingField.Date,
    };
}

export function SortingFieldToString(sf: SortingField): string {
    return SortingField[sf];
}